### DISCUSSION TITLE
Possible to update from EVO 1.4 to version 2.0 or to version 3.0?

### DISCUSSION THREAD

**Discussion by q2apro** [evolution-cms/evolution#1912] [Q&A]
Can I update from EVO 1.4 to version 2.0 or even to version 3.0 RC?

What is the best practice?

And is it possible to jump-upgrade from 1.4 to 3.0 RC?

<br>

I found the upgrade instructions at http://www.evolution-docs.com/documentation/administration/upgrading but it does not say something about the versions.

**Comment by fourroses666** [comment]
If I where you I would update from latest version of 1.4 to latest version of 2 (create config file..)

3 is not ready for production afaik

**Comment by q2apro** [comment]
When I upload Evolution CMS 2.0 and go to install, it says "Install a new copy of Evolution CMS - Please note this option may overwrite any data inside your database."

![image](https://user-images.githubusercontent.com/6764548/103400946-9f2da380-4b4f-11eb-889d-c7a99425134f.png)

So it cannot be updated/upgraded from 1.4 but needs a fresh installation?

Or even if I click "New Installation", is that data in the database untouched?

**Comment by fourroses666** [comment]
Yes you n ed to create a default.php file (config) in the core folder. Atm i can’t help you sorry need to go

**Comment by q2apro** [comment]
Seems that a fresh installation is necessary. 

In the installation process, when entering the prefix for the database tables (e.g. `modx_`), choosing the same prefix as the existing ones, gives:

> Checking database: failed - table prefix already in use!

Damn. I thought an update would be directly possible. 😔

**Comment by fourroses666** [comment]
I think there is an example file in root, but maybe someone else can help you. No time now

**Comment by fourroses666** [comment]
https://github.com/evolution-cms/evolution/tree/2.0.x/core/config/database/connections

There needs to be created a config file called default.php

But I have no example atm for you..

**Comment by Dmi3yy** [comment]
https://youtu.be/jbhrTlcjqQ4

See this video 

soon i will write new video 
1.4 to 3.0

**Comment by Dmi3yy** [comment]
https://github.com/evolution-cms/evolution/blob/3.x/install/stubs/files/config/database/connections/default.tpl

config file for 2.0 or 3.0

**Comment by q2apro** [comment]
@Dmi3yy It would be better to write a tutorial 1., 2., 3. ... in Markdown like `Readme.md` and then upload it to github (branch 1.4, for example having a folder `upgrade`).

Videos cannot be edited later on. Written tutorials can. If necessary, then adding images.

**Comment by Dmi3yy** [comment]
I think need both type

**Comment by fourroses666** [comment]
Did you manage to update it to 2.0?

Example file /core/config/database/connections/default.php
```
<?php
return [
    'driver' => env('DB_TYPE', 'mysql'), //$database_type
    'host' => env('DB_HOST', 'localhost'), //$database_server
    'port' => env('DB_PORT', '3306'),
    'database' => env('DB_DATABASE', 'site_db'), //$dbase
    'username' => env('DB_USERNAME', 'site_user'), //$database_user
    'password' => env('DB_PASSWORD', 'password'), //$database_password
    'unix_socket' => env('DB_SOCKET', ''),
    'charset' => env('DB_CHARSET', 'utf8mb4'), // $database_connection_charset
    'collation' => env('DB_COLLATION', 'utf8mb4_unicode_ci'), //$database_collation
    'prefix' => env('DB_PREFIX', 'evo_'), //$table_prefix
    'method' => env('DB_METHOD', 'SET NAMES'), //$database_connection_method
    'strict' => env('DB_STRICT', false),
    'engine' => env('DB_ENGINE'),
    'options' => [
        PDO::ATTR_STRINGIFY_FETCHES => true,
    ]
];
```

**Comment by fourroses666** [comment]
When you want to update to a newer version of EVO 2 I always need to change CHMOD of that file to 644.
Also QuickManager+ is active in plugins while I always set that unactive. These 2 things I always have to do before I update EVO 2.

**Comment by q2apro** [comment]
The folder `\core\config\database\connections\` is empty for me. 

Are you saying that EVO is doing only the update when this `default.php` file is there?

Otherwise EVO is doing a new installation?

**Comment by fourroses666** [comment]
Correct, config file is moved to an other place

**Comment by q2apro** [comment]
I have done the update and wanted to report back:

1. Download EVO 2 and unzip.
2. Rename folders: `assets` to `assets2`, `core` to `core2`, `manager` to `manager2` and file `index.php` to `index.php2`
3. Upload all files (including folder `install`) to server
4. Rename the older folders to: `assets` to `assets1`, `core` to `core1`, `manager` to `manager1` and file `index.php` to `index.php1`
5. From here your site will not work now
6. Rename the folders: `assets2` to `assets`, `core2` to `core`, `manager2` to `manager` and file `index.php2` to `index.php`
6. Navigate to yoursite.com/install
7. Do the "Upgrade" procedure
8. Navigate to yoursite.com/manager and login to verify the admin panel works

Notes:

For me, there were files/folders missing that I had in EVO1, so I copied them from EVO1 to EVO2:

- folder`/assets/plugins/filesource`
- folder `/assets/plugins/managermanager`

Done.

<br>

Another issue I faced: https://github.com/evolution-cms/evolution/issues/1750

**Comment by fourroses666** [comment]
The sites I upgrade from EVO 1.x to 2.x I do the following steps:

- make backup db + files.
- upload install.php in the root of the site -> https://github.com/evolution-cms/installer (or manually upload the EVO 2 files..)
- go to mysite.com/install.php or when manually upload the files mysite.com/install
- go true the first step of installing EVO 2 which uploads the files needed for installing.
- add the /core/config/database/connections/default.php file with the correct configuration (CHMOD 644). https://github.com/evolution-cms/evolution/issues/1745#issuecomment-753493990
- complete the installer (upgrade and not new installation)
- remove the .php files in the /assets/cache folder 
- login and check Extras for plugins which needs updates

**Comment by Dmi3yy** [comment]
https://www.youtube.com/watch?v=RjOiq35FgW4&t=1s
